### PR TITLE
flutter: 1.22.5 -> 1.22.4

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

PR #109477 bumped Flutter from 1.22.0 -> 1.22.4, and afterwards from 1.22.4 -> 1.22.5. Everything seems to work well (`flutter doctor` works), but trying to run `flutter create` results in:

```
$ flutter create project_test
...
Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
So, because project_test depends on flutter any from sdk, version solving failed.
pub get failed (server unavailable) -- attempting retry 1 in 1 second...
Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
So, because project_test depends on flutter any from sdk, version solving failed.
pub get failed (server unavailable) -- attempting retry 2 in 2 seconds...
Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
So, because project_test depends on flutter any from sdk, version solving failed.
pub get failed (server unavailable) -- attempting retry 3 in 4 seconds...
Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
So, because project_test depends on flutter any from sdk, version solving failed.
pub get failed (server unavailable) -- attempting retry 4 in 8 seconds...
Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
So, because project_test depends on flutter any from sdk, version solving failed.
```

I will try to investigate later on, but for now let's unbreak Flutter by unbumping it. But let's keep some of the changes after bump just to not break backwards compatibility thrice (even if the removal of `depsSha256` is a small breaking change).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by ericdallo — APPROVED** [review]

**Comment by thiagokokada** [comment]
@SuperSandro2000 ?

**Comment by SuperSandro2000** [comment]
> Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
> So, because project_test depends on flutter any from sdk, version solving failed.

What is this error message. I read it 5 times and still don't understand it.

File: pkgs/development/compilers/flutter/flutter.nix [hunk 1]
```diff
@@ -70,6 +65,9 @@ let
       ${dart}/bin/dart --snapshot="$SNAPSHOT_PATH" --packages="$FLUTTER_TOOLS_DIR/.packages" "$SCRIPT_PATH"
       echo "$revision" > "$STAMP_PATH"
       echo -n "${version}" > version
+
+      rm -rf bin/cache/{artifacts,dart-sdk,downloads}
+      rm -f  bin/cache/*.stamp
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/flutter/flutter.nix:70]:
```suggestion
      rm -r bin/cache/{artifacts,dart-sdk,downloads}
      rm  bin/cache/*.stamp
```

**Comment by thiagokokada** [comment]
> > Because every version of flutter from sdk depends on sky_engine any from sdk which is forbidden, flutter from sdk is forbidden.
> > So, because project_test depends on flutter any from sdk, version solving failed.
> 
> What is this error message. I read it 5 times and still don't understand it.

I also don't know, but my suspect is that the `sky_engine` dependency from Flutter is conflicting with `sky_engine` dependency of the project itself (don't know why this is happening).

But it is quite easy to reproduce, go to master and do `flutter create test_project`.

**Comment by SuperSandro2000** [comment]
> But it is quite easy to reproduce, go to master and do flutter create test_project.

I have zero knowledge about flutter so I can't help you in any meaningful way.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>flutter (flutterPackages.stable)</li>
<li>hover</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
@ericdallo it would be nice if you could take a look at the refactor.

other than that I the change looks fine to me but I can't tell if it is the right decision.

**Review by ericdallo — APPROVED** [review]
To fix the failing derivation LGTM

**Pull request merged**